enable custom css modules compilers

nathantreid opened the pull request:
I'd like to add support for custom CSS modules compilers similar to your support for custom "lang" processors. If there is no custom processor, it defaults to the basic processor I added in #117.
The main reason for this PR is so I can interop with my css-modules build compiler, allowing the user to import other CSS modules files and use all of the configuration options I provide.

The compile call is identical to the one for lang compilers with the addition of passing in the `styleTag` and sourcemap. I pass in the `styleTag` because I need to set the `autoprefix` attribute to `off` if the user is already using autoprefixer through my package as well as read the `module` attribute so I can implement support for multiple style tags. _I'll need to make another PR eventually since the module attribute should also allow [overriding the computed property name](http://vue-loader.vuejs.org/en/features/css-modules.html)._

You'll also note that in handling the result I append to the `js` variable - this is because CSS modules reference other files via composes (as opposed to including them in the source like sass imports). My plugin therefore outputs `require` statements so Meteor will load those referenced files.

I wasn't sure how exactly you'd want the cssModules compiler to be specified so for now I used `global.vue.cssModules`.

Akryum commented:
Would you mind documenting these changes? 😄 
Awesome work BTW.

nathantreid commented:
Documentation?! The horror! Yeah, I'll write this up.

nathantreid commented:
I'm hoping to knock this out this weekend, and just wanted to clarify what I'm documenting. So far I've got:
1. document the `module` attribute in the vue-component documentation (I'm adding it after the scoped styles section).
2. mention my community package (which I have to release before I submit this documentation!) 😄 

I was planning to document adding your own custom css modules compiler, but I'm not sure where I should put that since I don't see explicit instructions for others (like the `lang` processors) - I just see pointers to the actual packages.
So before I go much further, what would you like me to document?

Akryum commented:
You could add the package to the "lang" list, and then document in its README what it does and how to use the features it add to the Vue components.

nathantreid commented:
I found a logging statement leftover from my initial CSS modules PR, so I removed that.
I also added a section on CSS modules usage via the  `module` attribute and put a reference to my package (https://github.com/nathantreid/vue-css-modules) in the community packages section.

Akryum commented:
Awesome! Thank you for your work!
